portfolio/src/main/webapp/script.js

```diff
@@ -26,5 +26,28 @@ async function requestPuzzle() {
 
     //here it should be rendered to the page
     console.log(responseObject);
+    
+    const puzzlePage = document.getElementById("thisPuzzle");
+    const puzzleDifficulty = responseObject.difficulty;
+    const puzzleImage = responseObject.imageUrl;
+
+    if (puzzleDifficulty == "EASY"){
+        puzzlePage.innerHTML = `<img src=${puzzleImage} onLoad="snapfit.add((this), {level:1, mixed:true, snap:12, nokeys:true, aborder:true, polygon:true, aimage:true, aopacity: 0.3, space:15, callback:function(){solvedPuzzle()} });" width="750">`;
+    }
+
+    if (puzzleDifficulty == "MEDIUM"){
+        puzzlePage.innerHTML = `<img src=${puzzleImage} onLoad="snapfit.add((this), {level:3, mixed:true, snap:12, nokeys:true, aborder:true, polygon:true, aimage:true, aopacity: 0.3, space:15, callback:function(){solvedPuzzle()} });" width="750">`;
+    }
+
+    if (puzzleDifficulty == "HARD"){
+        puzzlePage.innerHTML = `<img src=${puzzleImage} onLoad="snapfit.add((this), {level:5, mixed:true, snap:12, nokeys:true, aborder:true, polygon:true, aimage:true, aopacity: 0.3, space:15, callback:function(){solvedPuzzle()} });" width="750">`;
+    }
+
+    //link to image download after solved
+    document.getElementById("downloadImg").innerHTML = `<a href="${puzzleImage}">Click here to download image!</a>`;
   }  
 }
+
+async function solvedPuzzle(){
```

Comment:
the `async` can be removed here because the function doesn't `await` any asynchronous functions.